yang-module-versioning/draft-ietf-netmod-yang-module-versioning.xml

```diff
@@ -772,27 +764,39 @@ module ietf-yang-revisions {
   // RFC Ed.: replace XXXX (inc above) with actual RFC number and
   // remove this note.
 
-  revision 2022-10-10 {
+  revision 2022-11-21 {
     rev:revision-label 1.0.0-draft-ietf-netmod-yang-module-versioning-07;
     description
       "Initial version.";
     reference
       "XXXX: Updated YANG Module Revision Handling";
   }
 
+  typedef revision-date {
+    type string {
+      pattern '[0-9]{4}-(1[0-2]|0[1-9])-(0[1-9]|[1-2][0-9]|3[0-1])';
+    }
+    description
+      "A date associated with a YANG revision.
+
+      Matches dates formatted as YYYY-MM-DD.";
+    reference
+      "RFC 3339: Date and Time on the Internet: Timestamps";
+  }
+
   typedef revision-label {
     type string {
       length "1..255";
       pattern '[a-zA-Z0-9,\-_.+]+';
-      pattern '\d{4}-\d{2}-\d{2}' {
+      pattern '[0-9]{4}-(1[0-2]|0[1-9])-(0[1-9]|[1-2][0-9]|3[0-1])' {
```

Comment:
I'd suggest we instead make this a union of the string with our new revision-date typedef.

Reply:
Should the invert-match of the revision-date pattern just be removed then?

Reply:
I mean, it's confusing, but not illegal (right?), to have a revision-label
formatted as a YYYY-MM-DD date.

Reply:
Ahh. I didn't notice the invert match :-). I think we probably do want to keep this in then. Some places use a revision-label-or-date and we need to recognize which it is.

Reply:
Add a plain text description about the pattern invert match in the YANG module (maybe in error message of pattern)